CI/deploy-mudlet-for-windows.sh

```diff
@@ -0,0 +1,340 @@
+  mv "${PACKAGE_DIR}/$uploadFilename" "$uploadDir/"
+  
+  if [[ "$uploadFilename" == *.zip ]]; then
+    $uploadFilename="${uploadFilename%.zip}"
+  fi
+
+  # Append these variables to the GITHUB_ENV to make them available in subsequent steps
+  echo "FOLDER_TO_UPLOAD=${uploadDir}\\" >> "$GITHUB_ENV"
+  echo "UPLOAD_FILENAME=$uploadFilename" >> "$GITHUB_ENV"
+}
+
+# Check if GITHUB_REPO_TAG and PublicTestBuild are "false" for a snapshot build
+if [[ "$GITHUB_REPO_TAG" == "false" ]] && [[ "$PublicTestBuild" == false ]]; then
+  echo "=== Creating a snapshot build ==="
+  mv "$PACKAGE_DIR/mudlet.exe" "Mudlet.exe"
+  
+  # Create a zip file using 7z
+  7z a "Mudlet-$VERSION$MUDLET_VERSION_BUILD-$BUILD_COMMIT-windows-$BUILD_BITNESS.zip" "$PACKAGE_DIR/*"
+
+  # Define the upload filename
+  uploadFilename="Mudlet-$VERSION$MUDLET_VERSION_BUILD-$BUILD_COMMIT-windows-$BUILD_BITNESS.zip"
+
+  # Move packaged files to the upload directory
+  moveToUploadDir "$uploadFilename"
+else
+
+  # Check if it's a Public Test Build
+  if [[ "$PublicTestBuild" == "true" ]]; then
+
+    # Get the commit date of the last commit
+    COMMIT_DATE=$(git show -s --format="%cs")
+    # Get yesterday's date in the same format
+    YESTERDAY_DATE=$(date --date="yesterday" +%Y-%m-%d)
+
+    if [[ "$COMMIT_DATE" < "$YESTERDAY_DATE" ]]; then
+      echo "=== No new commits, aborting public test build generation ==="
+      exit 3
+    fi
+
+    echo "=== Creating a public test build ==="
+    # Squirrel uses Start menu name from the binary, renaming it
+    mv "$PACKAGE_DIR/mudlet.exe" "$PACKAGE_DIR/Mudlet PTB.exe"
+    # ensure sha part always starts with a character due to a known issue
+    VersionAndSha="${VERSION}-ptb-${BUILD_COMMIT}"
+
+  else
+    echo "=== Creating a release build ==="
+    mv "$PACKAGE_DIR/mudlet.exe" "$PACKAGE_DIR/Mudlet.exe"
+    VersionAndSha="$VERSION"
+  fi
+
+  echo "=== Cloning installer project ==="
+  git clone https://github.com/Mudlet/installers.git "$GITHUB_WORKSPACE/installers"
+  cd "$GITHUB_WORKSPACE/installers/windows" || exit 1
+
+  echo "=== Installing Squirrel for Windows ==="
+  nuget install squirrel.windows -ExcludeVersion
+
+  echo "=== Setting up directories ==="
+  SQUIRRELWIN="$GITHUB_WORKSPACE/squirrel-packaging-prep"
+  SQUIRRELWINBIN="$SQUIRRELWIN/lib/net45/"
+
+  if [[ ! -d "$SQUIRRELWINBIN" ]]; then
+    mkdir -p "$SQUIRRELWINBIN"
+  fi
+
+  echo "=== Moving things to where Squirrel expects them ==="
+  mv "$PACKAGE_DIR/"* "$SQUIRRELWINBIN"
+
+  # Set the path to the nuspec file
+  NuSpec="$GITHUB_WORKSPACE/installers/windows/mudlet.nuspec"
+  echo "=== Creating Nuget package ==="
+
+  # Check if this is a Public Test Build
+  if [[ "$PublicTestBuild" == "true" ]]; then
+    # Allow public test builds to be installed side by side with the release builds by renaming the app
+    # No dots in the <id>: Guidelines by Squirrel
+    sed -i "s/<id>Mudlet<\/id>/<id>Mudlet_${BUILD_BITNESS}_-PublicTestBuild<\/id>/" "$NuSpec"
```

Comment:
No. You must **not** use a different Id for the 32-bit builds compared to what we were using before - otherwise existing installed Mudlet PTB versions will not see this as an update. For those we were using:
**`Mudlet-PublicTestBuild`** as the id and that is what it must stay as.

Reply:
Err, I understood that the 32 bit should have remained the same.. I guess i meant to wrap that in a test for MINGW64 but forgot to before I did a commit, oopsie